code/modules/mapping/space_management/space_reservation.dm

```diff
@@ -1,35 +1,139 @@
+#define FIXED_RESERVE_HORZ_PADDING 9
+#define FIXED_RESERVE_VERT_PADDING 9
+#define FIXED_RESERVES_HORIZONTAL 2
+#define FIXED_RESERVES_VERTICAL 2
+#define FIXED_RESERVE_WIDTH round((world.maxx - FIXED_RESERVE_HORZ_PADDING * (FIXED_RESERVES_HORIZONTAL-1))/FIXED_RESERVES_HORIZONTAL)
+#define FIXED_RESERVE_HEIGHT round((world.maxy - FIXED_RESERVE_VERT_PADDING * (FIXED_RESERVES_VERTICAL-1))/FIXED_RESERVES_VERTICAL)
 
 //Yes, they can only be rectangular.
 //Yes, I'm sorry.
 /datum/turf_reservation
-	var/list/reserved_turfs = list()
 	var/width = 0
 	var/height = 0
 	var/bottom_left_coords[3]
 	var/top_right_coords[3]
-	var/wipe_reservation_on_release = TRUE
-	var/turf_type = /turf/open/space
-	var/border_turf_type
-	var/area/area_type
 	var/virtual_z_level
 
-/datum/turf_reservation/transit
-	turf_type = /turf/open/space/transit
+/datum/turf_reservation/Destroy()
+	SSmapping.reservations_by_level["[bottom_left_coords[3]]"] -= src
+	return ..()
 
-/datum/turf_reservation/proc/Release()
-	var/v = reserved_turfs.Copy()
-	for(var/i in reserved_turfs)
-		reserved_turfs -= i
-		SSmapping.used_turfs -= i
-	SSmapping.reserve_turfs(v)
+/datum/turf_reservation/proc/reserve()
+	SHOULD_NOT_OVERRIDE(TRUE)
+	if(width > world.maxx || height > world.maxy)
+		CRASH("Cannot create turf reservation of size [width] width, [height] height; world size is only [world.maxx] wide by [world.maxy] tall!")
+	if(width < 1 || height < 1)
+		CRASH("Cannot create turf reservation of size [width] width, [height] height; both dimensions must be 1 or greater!")
+
+	for(var/z_level in SSmapping.levels_by_trait(ZTRAIT_RESERVED))
+		if(reserve_on_z(z_level))
+			return TRUE
+
+	// If we didn't return at this point, theres a good chance we ran out of room on the exisiting reserved z levels, so lets try a new one
+	var/datum/space_level/new_reserved = SSmapping.new_reserved_level()
+	if(reserve_on_z(new_reserved.z_value))
+		return TRUE
+
+	// fuck
+	return FALSE
+
+/datum/turf_reservation/proc/reserve_on_z()
+	PROTECTED_PROC(TRUE)
+	SHOULD_CALL_PARENT(TRUE)
+	SSmapping.reservations_by_level["[bottom_left_coords[3]]"] |= src
+	virtual_z_level = get_new_virtual_z()
+	return TRUE
+
+/datum/turf_reservation/proc/get_reserved_turfs()
+	SHOULD_NOT_OVERRIDE(TRUE)
+	var/turf/bottom_left = locate(bottom_left_coords[1], bottom_left_coords[2], bottom_left_coords[3])
+	var/turf/top_right = locate(top_right_coords[1], top_right_coords[2], top_right_coords[3])
+	return block(bottom_left, top_right)
+
+/datum/turf_reservation/proc/get_non_border_turfs()
+	SHOULD_NOT_OVERRIDE(TRUE)
+	var/turf/bottom_left = locate(bottom_left_coords[1]+1, bottom_left_coords[2]+1, bottom_left_coords[3])
+	var/turf/top_right = locate(top_right_coords[1]-1, top_right_coords[2]-1, top_right_coords[3])
```

Comment:
When we add padding for mesons and such (if we do at all) we need to remember to replace +/- 1 with a #define-d or variable amount

Reply:
they're already padded, it's integrated into the reserve sizes themselves -- the +/- 1 is to account for the 1-tile impassable border, which is included inside the reserve because it's placed on the tiles of the reserve